### DISCUSSION TITLE
Busch-Jaeger once more

### DISCUSSION THREAD

**Discussion by afoeder** [Koenkk/zigbee2mqtt#7009] [Devices]
There is still some unadressed issues with the Busch-Jager [6735/6736/6737](https://www.zigbee2mqtt.io/devices/6735_6736_6737.html); to get a gist check this [archived discourse thread](https://zigbee2mqtt.discourse.group/t/some-issues-trying-to-get-busch-jaeger-to-work/768) that adresses #879.

The double-row items so far work as intended on a 6710 U, 6711 U and 6715 U (i.e. the power supply device, the relais device and the dimmer device; in the meantime I own all of them).
However, when trying to use the four-row control panel on a 6710, I can observe the _actions_ for the top three rows (i.e. 
`row_1_off`, `row_1_down`, `row_1_stop`, `row_1_on`, `row_1_up` etc).
The fourth row doesn't emit and their LED indicator blink when pressing, while the first three who do emit actions just briefly light a single time to indicate acknowledgement.

It feels like the fourth row is still in some scene thingie mode.

Is there some swarm knowledge here in the meantime who got any further information? ~Unfortunately the main contributor of the Busch-Jaeger devices @docbobo seems to be not available anymore :(~ // edit: thanks @docbobo!

**Comment by docbobo** [comment]
The blinking means that the row is not bound to anything. In that mode, you should be able to store a particular scene on the button, as described in the manual. Luckily for you, I just had the same problem when I was looking into what you were just describing.

The problem with the Busch Jaeger devices is that they have very limited memory. With the two or one row versions that hardly an issue, but with the 4 row version it's a problem.

Anyway, I had the same problem and the zigbee2mqtt UI was showing correct bindings. I removed the binding for Source EP 13 (row 4), and then created a new one binding Source EP 13 to Coordinator/LevelCtrl. Blinking stopped. I also see row_4_on and row_4_off events in HA.

BTW: You should be able to also unbind any other row to get into the "scene programming" mode. If you unbind e.g. EP 11 (2nd row), it should be possible to store a scene as described in the switches manual.

**Reply by afoeder** [reply]
oh thanks a lot! I've mentioned you here and there in other issues without luck, therefore I assumed you were "lost" :) 

I gotta admit that I don't fully understand what you're saying yet, I seem to lack some basic (protocol) knowledge. Could you please elaborate on particularly

* > zigbee2mqtt UI was showing correct bindings
    
    where can I see those bindings? I checked the [doc page](https://www.zigbee2mqtt.io/information/binding.html) but I cannot find anything related in the UI
* > Source EP […] Coordingator/Level Ctrl
    
    what's EP? and what's Coordinator/Level Ctrl?
    
    
Thanks for taking the time to explain it to me (feel free to add some screenshots, that'd be highly appreciated); and I commit on adding this information to the respective Doc page (like a new chapter "common issues" for the Busch-Jaeger items).

**Comment by docbobo** [comment]
I am definitely not gone for good but the time required for things like this has been lost to other important stuff. That said, @Koenkk is probably much better suited to explain all the details of zigbee, so I'll limit my response to what I believe is need to help you out.

Every zigbee device exposes its capabilities through a set of endpoints (EP). The capabilities themselves are grouped into clusters and each device indicates the clusters that are acceptable as either input or output of a particular endpoint. For example, a 6737 running on an in-wall power adapter expose endpoints 10, 11, 12, 13 which represent the individual rows on that switch. 

Binding sets up a relation ship between two different endpoints on different devices or groups. For example, you could bind the OnOff cluster of ep 10 (on a 6737) to the OnOff cluster of endpoint 11 of a hue bulb. As a result, you can use the first row of the 6737 to turn the hue bulb on and off, even if Home Assistant is not running.

Now back to your problem. When you click on the Busch Jaeger device in the zigbee2mqtt UI, you'll get something similar to this:

<img width="1006" alt="Screen Shot 2021-04-09 at 10 37 49" src="https://user-images.githubusercontent.com/1257060/114153570-bbe03100-991f-11eb-81e0-1d7e01eb1fde.png">

Clicking on the "Bind" tab will lead to this:

<img width="1014" alt="Screen Shot 2021-04-09 at 10 39 13" src="https://user-images.githubusercontent.com/1257060/114153690-e205d100-991f-11eb-8c9d-d0ec23102b4f.png">

There is a row for Source EP 13, which corresponds to the fourth row on the switch. 
- Click on unbind to remove the binding - the row should disappear. 
- Afterwards, select 13 in the Source EP field at the bottom, select Coordinator as Destination. Keep the Destination EP at 1 but select LevelCtrl as Cluster. 
- Press the blue bind button.

If binding is successful a new row with the binding should appear. Theoretically, things should work at this point.

BTW: Some of the things I described in the beginning are nicely visualized under the Cluster tab.

**Reply by afoeder** [reply]
woah this makes so much sense now!! thank you! So, the binding information that is present "by default", is this actually done by the [code for the converters](https://github.com/Koenkk/zigbee-herdsman-converters/blob/8299a69deee2fa7ac4ddb6667bc2e3444a963be9/devices.js#L10716-L10764)?

And given the steps you have described, am I right that I just "recreate" the binding which is already present? Having said this, why doesn't it work "initially"? Why do I have to re-bind? (not blaming of course, just trying to understand).

Thanks 💯  again! 

// edit: I did that "re-binding" but with no luck… the bottom row is still blinking when pushing the buttons… can I try something "more"?

**Reply by docbobo** [reply]
Yes, that's the code that is performing the initial setup. To be honest, I don't know why this code stopped working correctly - I am pretty sure it did when it was first committed (I hope this doesn't sound like an excuse). Maybe some internal timing has changed.

Given my general experience with those Busch Jaeger devices, the behavior is mostly likely caused by the devices themselves. For example, long before I was using zigbee2mqtt, the devices were regularly losing information about the stored scenes and I had to reprogram them every few month - at least some of them. Based on that and the limitation on the number of stored bindings, I suspect that there Is just some race condition.

Theoretically, you could try to see how reproducible that issue is when you unpair and pair the device. If it is, you could reverse the loop in the code above. As a result, I'd expect a different row to be bound incorrectly. If it is a timing problem, maybe it can be solved easily.

@Koenkk: I believe zigbee2mqtt does a few things in parallel, depending on the coordinator hardware. Would it try to send all those bindings in parallel in such a case?

**Reply by docbobo** [reply]
@afoeder can you try recreating all the bindings? To be honest, I did that by mistake earlier today. If that works, I might have an idea...

**Reply by afoeder** [reply]
I skipped your ideas with reversing the loop and repairing the device and directly tried recreating all the bindings manually one after another via the GUI and this worked! 🥳  Now all four rows are emitted!
Having said this, do you think it would help to add a 300ms-ish delay in the binding code when pairing the devices?

For the background, is there some doc/literature on what the "Coordinator" is and what the Clusters refer to? Are the clusters defined by zigbee (protocol) or is it an z2m "invention"?

> (I hope this doesn't sound like an excuse)

don't you dare to excuse for what you did in your free-time and what helps other guys who gave an arm and a leg to unleash capabilities of these overpriced Busch-Jaeger devices! :)

**Reply by afoeder** [reply]
btw it's "funny" (weird?) that Busch-Jager decided to not emit the actions of the top row when there's a Dimmer or Relais attached: just why not doing it anyways, it won't hurt if it's not listened to…

**Reply by docbobo** [reply]
First of all: glad to hear that it's working.

Having thought about this further, I believe that this is related to the comment that you can also see in the code:

```
// The total number of bindings seems to be severely limited with these devices.
// In order to be able to toggle groups, we need to remove the scenes cluster
const index = endpoint.outputClusters.indexOf(5);
if (index > -1) {
    endpoint.outputClusters.splice(index, 1);
}
```

I have this nagging feeling that this was added specifically to address the issue with the fourth row not being configured properly. However, it stopped to work exactly the way it's supposed to work since then, but once you recreate the binding it takes effect. Not sure if this timing related to just due to internal code changes.

And yes, it's completely annoying that dimmers and relays don't emit events on the first row. That's why I replaced almost all of my relais...

**Reply by Joe3n** [reply]
@docbobo  Could you post a picture of your binding of a 6711U, preferable with a 6735 (if you have it). I have some problems with it like descripted in Ticket https://github.com/Koenkk/zigbee2mqtt/issues/7077

![grafik](https://user-images.githubusercontent.com/32760573/116881270-56801700-ac23-11eb-9447-a7f33bf3bd0e.png)

**Reply by blib-blub** [reply]
Sorry for necroing a nearly 4 year old thread, but it seems I somehow managed to get my BJ 6737 (4 rockers) to emit zigbee events on the first row despite being on a 6711U Relais and no longer causing Relais switching on the 6711U. I am currently using them in Deconz managed by HA and I am in the process of migrating to Z2M. 

At some point I rewired some of the Relais outputs to be no longer connected and hardwired the lights to be permanently supplied. (Son likes to push the wrong buttons...  ;)). It does not work on all of the rewired switches, so I am not sure if the order of rewiring / pairing or some other thing caused them to behave that way. But there seems to be some mode that the switches can enter to be "fully Zigbee" even on a 6711 Relais.  Do you maybe know anything about this?

**Reply by TSim001** [reply]
> Sorry for necroing a nearly 4 year old thread, but it seems I somehow managed to get my BJ 6737 (4 rockers) to emit zigbee events on the first row despite being on a 6711U Relais and no longer causing Relais switching on the 6711U. I am currently using them in Deconz managed by HA and I am in the process of migrating to Z2M.
> 
> At some point I rewired some of the Relais outputs to be no longer connected and hardwired the lights to be permanently supplied. (Son likes to push the wrong buttons... ;)). It does not work on all of the rewired switches, so I am not sure if the order of rewiring / pairing or some other thing caused them to behave that way. But there seems to be some mode that the switches can enter to be "fully Zigbee" even on a 6711 Relais. Do you maybe know anything about this?

Hi blib-blub,
have you found the cause of the behavior you described?  Specifically, why the first row of your 6737 relays worked even when you were using the 6711u relay? It would be great to be able to retrieve the events from the first row. To keep track of the relay's current state, I'm currently using MQTT publishing. I send a status query to the Busch-Jaeger devices every 5 seconds to get their relay status. But this isn't an ideal solution. It would be great if you could briefly reply and let me know if you've found a solution. Thanks!

**Comment by FloW990** [comment]
Hi there,

I have the same problem that afoeder had. (the buttons in row four do not work).
Unfortunately, I am not very knowledgeable about programming. 
I would like to use z2m in combination with node-red.
Now I have already tried the z2mAssistant in use with Docker, unfortunately the server is displayed there as not connected. (although the top three buttons go into node-red at the same time?).
I would be very happy if the exact procedure for the new binding of the buttons would be explained here again. (gladly for dummies) :)
Thank you for the fantastic work everyone has done here since then
Greetings Flo

**Comment by cerisemagenta** [comment]
Hi, I'm trying to use my Busch Jaeger 6736 on my HA running zigbee2mqtt as an add-on. I'm using a ZONOFF CC2652P Dongle.
Z2M is identifying the switch but can only find one endpoint (10) which I understood is the upper row which doesn't emit any zigbee event. The lower row (which should be 11) is not identified at all and hence I don't get any events in z2m.
Any ideas what I could do?

Swich is running on firmware 1.0.0 - I read that there is a BJ update which however requires BJ support to send a stick + update file...

**Reply by klada** [reply]
This old firmware might definitely cause issues. I wonder whether OTA updating through Z2M would work.

Could you please show a screenshot of the clusters your switch is advertising? Does it have a genOta cluster?

**Reply by fdfdgsgsdfas** [reply]
Hi, I have a Busch Jäger Zigbee Light Link 6736 and just want to share a screenshot of my cluster with the genOta cluster. Would be great to get an OTA functionality. My switch has the Version 1.2.0
<img width="1080" height="1943" alt="Screenshot_20251107-114440" src="https://github.com/user-attachments/assets/cd0d2c57-e051-41c7-912a-52c781a2b971" />

**Comment by blib-blub** [comment]
I found out that some of the pins of the 6737 got bent when installing and where shorted, unsure which ones though. But I guess voltage/resistance on those pins is how 6737 knows on which base it sits.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by TSim001** [comment]
First of all, thank you very much for all the helpful information.

@docbobo @klada and Community:
Could you tell me/ us if there's a solution or workaround for the following two topics by now?

Topic 1:
Is it possible in any way to control the first row of Busch-Jaeger 6735/6736/6737 switches in conjunction with a 6711U (relay) or 6715U (dimmer)? In other words, to control the first row like rows 2 through 4 with their different states? If so, how can this be achieved? The description above, including linking the first row to endpoint 10, doesn't work. However, I'm also not sure if these instructions were only intended for the version with a 6710 (power supply) or battery operation.

In this context, I also find blib-blub's suggestion interesting. But I'd rather not "bend" or remove the pins or modify the hardware. Experts could probably assess this better.

Topic 2:
Is it only possible to obtain the status of the 6711U relay via polling? Unfortunately, the polling mechanism for the Busch-Jaeger switches in Zigbee2MQTT always generates two MQTT publish messages per Busch-Jaeger switch in my setup, which puts a strain on the network. Therefore, I'm currently using my own automation (MQTT publishing) to receive the status every 5 seconds. Or do you perhaps have another solution for this?

I know these two topics have already been discussed in several forum posts. But some of those posts are already five years old and often contain contradictory information. Perhaps your answer will help to summarize the current situation for everyone who is currently using Busch-Jaeger switches. Thank you for your reply and help.

My setup:
Zigbee2MQTT on HA with the latest versions
Busch-Jaeger 6736/6737 with version 1.2.0

**Comment by docbobo** [comment]
@TSim001 TBH, I am not tinkering a lot with those devices these days. I've especially given up on the idea of detecting the state of devices controlled via the relay in realtime. Every time I thought I understood how it's working, it just stopped doing so. 

Topic 1: I believe you are asking if you can bind the first row so that it can be used to either control other devices or at least to emit events in realtime. As far as I know, and based on the tests I did in the past, that is not possible. The moment you attach the rocker to a relay, endpoint 10 basically becomes useless. Unfortunately, the devices still reports it; however, I don't recall ever seeing a message emitted for that cluster in that setup. 

For that reason, I replaced one or two of my relays with just the power supply, adding an actor to the actual device I wanted to control. For example, I have an illuminated mirror in the restroom that I used to control via the relay - now the mirror itself has its own actor.

Topic 2:
Similar to topic 1, I don't believe this is possible. To be fully transparent: this has been my hope in the beginning as well and was the main reason for switching to zigbee2mqtt for me. Right now, my relays are set to a default polling interval of 60s, which doesn't add too much load to the network. But yes, trying to do that every 5 seconds is probably adding lots of strain, especially if you have lots of relays. If you want to avoid that, your best option would probably be to replace the relay with a power supply/actor combination as mentioned above. If you have enough space in your junction box, you might even be able to do it directly there.

Out of curiosity: are you concerned about the two MQTT messages? Or rather the idea that they are caused by too much zigbee traffic?

**Reply by TSim001** [reply]
@docbobo 
Thanks for your quick and detailed reply. :) Are you still using the Busch-Jaeger relays and switches that we're discussing here?

Generally, I find the Busch-Jaeger control elements very good for many use cases and they integrate perfectly into the existing Busch-Jaeger Future Linear switch/socket series. They have all the functions of a remote control and are also installed directly as flush-mounted switches. With other solutions, like the Sonoff or Shelly flush-mounted relays, you only have the standard on/off switching function and no special actions like Hold/Hold-Released for dimming. And, most importantly, they don't offer the 8 button functions of the 6737. Or do you know of any other product that offer the same functions as the 6737 and are also compatible with the Busch-Jaeger Future Linear series?

Now to the two topics:
I already suspected that there wasn't a (successful) solution for either of them yet. Thanks for confirming that. I had also considered using the 6710u instead of the 6711u. However, I didn't do it because if there's a problem with Zigbee2MQTT, the 6737 won't be able to control the lights. Or is it possible to solve this with a direct binding between the 6737 and the lights? So that the Busch-Jaeger addresses the lights directly via the endpoint/group? Does the direct connection impose any limitations in Zigbee2MQTT or Home Assistant?

Regarding your question about the MQTT messages:
I have the following problem: If I enter, for example, 5 seconds under "State poll Interval" in Zigbee2MQTT for the 6737, I always get duplicate log entries/ status messages for the status of the 6737 (i.e., 2 x 2 log entries / messages per device). If I handle the MQTT polling via a separate automation, I only get one log entry (2 messages per device).

To avoid overloading the Zigbee network with duplicate Zigbee status messages, I'm now using my automation instead of polling via the Zigbee2MQTT configuration. Another advantage of my automation is that I can control the polling myself. For example, I can request 5 devices, then pause for 2 seconds, then request the next 5 devices, and so on. This further reduces the load on the Zigbee network regarding parallel requests.

Unfortunately, 60 seconds would be too long for me. I'd like to see the current status of the light or relay in the dashboard, and that's only possible if it's updated promptly.

One more question I'm interested in:
Do you also use the relays to switch power outlets, or just lights? I'm unsure about power outlets because of the maximum load of approximately 2300W. If a high-power device is plugged into a power outlet, the relay could potentially be damaged if it's switched.

**Reply by docbobo** [reply]
Let me quickly describe how I am using those things:

In my office I am using a a 6736 - a two row panel - together with a 6710U. In addition to the default endpoints, endpoint 10 - the first row - is bound to a group called `office_all` using the `genOnOff` cluster. As a result, all devices that are part of the group toggle via the 6736, independent of whether zigbee2mqtt is running or not.

In addition, zigbee2mqtt will emit events like `on_row_1` and `off_row_1` right away. It will do the same for the second row.

I have NodeRED running with a zigbee2mqtt integration (I believe). I've created some flow that when pressing the On button again, it will cycle through the scenes I have set up via zigbee2mqtt as well. But you can do whatever you like there. 

Now, I've noticed in the past that sometimes I had to completely re-add the 6736 when trying to adjust the members of the group but that's been a rather small price to pay. I do agree that the control elements integrate well with the Future Linear and I haven't found anything better either.

Lastly, I do not switch power outlets via the relay. However, that I described above should work well with zigbee power plugs, so you could do that instead.

**Reply by docbobo** [reply]
Oh, changing the binding might also require you to remove and re-add the switch. Had that happen in the past as well.
